docker-stubs/api/container.pyi

```diff
@@ -9,145 +8,131 @@ from ..types import (
     HostConfig as HostConfig,
     NetworkingConfig as NetworkingConfig,
 )
-from _typeshed import Incomplete
-
 from ..types.base import Command, PathStr, MacAddress, Signal
 
-
-class CreateContainerReturnDict(TypedDict):
+class CreateContainerReturndict(TypedDict):
     Id: str
     Warnings: list[str]
 
-
 class ContainerApiMixin:
     def attach(
         self,
-        container,
-        stdout: bool = ...,
-        stderr: bool = ...,
-        stream: bool = ...,
-        logs: bool = ...,
-        demux: bool = ...,
-    ): ...
+        container: str,
+        stdout: bool = True,
+        stderr: bool = True,
+        stream: bool = True,
+        logs: bool = True,
+        demux: bool = False,
+    ) -> CancellableStream: ...
     def attach_socket(
-        self, container, params: Incomplete | None = ..., ws: bool = ...
-    ): ...
+        self, container: str, params: Optional[dict[str, str]] = None, ws: bool = False
+    ) -> CancellableStream: ...
     def commit(
         self,
-        container,
-        repository: Incomplete | None = ...,
-        tag: Incomplete | None = ...,
-        message: Incomplete | None = ...,
-        author: Incomplete | None = ...,
-        pause: bool = ...,
-        changes: Incomplete | None = ...,
-        conf: Incomplete | None = ...,
-    ): ...
+        container: str,
+        repository: Optional[str] = None,
+        tag: Optional[str] = None,
+        message: Optional[str] = None,
+        author: Optional[str] = None,
+        pause: bool = True,
+        changes: Optional[list[str]] = None,
+        conf: Optional[ContainerConfig] = None,
+    ) -> dict[str, str]: ...
     def containers(
```

Comment:
do we really want to include the default arguments here? I'm worried they'll fall out of step with the actual values in the library

Reply:
I believe there is a `= ...` syntax option now to show in a stub that there are default arguments

Reply:
Why not use the new optional syntax? And yes you are not supposed to put the actual for the default but `...`.

Reply:
there's new optional syntax?